### ISSUE TITLE
window and variance

### ISSUE THREAD

**Issue by JuliaLawall** [dacapobench/dacapobench#192]
The help offers the following options:

```
    --variance <pct>                          Target coefficient of
                                              variation <pct> (default
                                              3.0)
    --window <window>                         Measure variance over
                                              <window> runs (default 3)
```
But I don't see any evidence that these options have any impact: I always get only one run, unless I use -n.  For example, no matter what the window size, I always get the same overall execution time.  How are these arguments supposed to work?

**Comment by steveblackburn** [comment]
Hi Julia,

I'll need to investigate.    They certainly _did_ work.   What version are you using?

I've been focussed 100% on the next release, and have not yet tested these for it.   They did work with prior releases; at least I'm unaware until now of any issues.

--Steve

**Comment by JuliaLawall** [comment]
On Tue, 10 Aug 2021, Steve Blackburn wrote:

>
> Hi Julia,
>
> I'll need to investigate. They certainly did work. What version are you using?
>
> I've been focussed 100% on the next release, and have not yet tested these for it. They did work with prior releases; at least I'm unaware until now of any issues.

I using Debian 11.  My .jar file is dacapo-9.12-MR1-bach.jar

I was also trying the "evaluation" benchmarks
(dacapo-evaluation-git+309e1fa.jar), but I started noticing that
a number of them were generating various complaints about reflection, and
other things, so I got the impression that the whole thing was not
reliable.  So it seemed safer to take the ones thta had been chosen by the
Phoronix test suite, which are the old ones.  If you are confident that
the evaluation benchmarks actually work on Debian 11, ie actually have
their proper fonctionality despite the various complaints, then I would
definitely consider using them instead.

thanks,
julia

**Comment by steveblackburn** [comment]
Hi Julia,

Thanks for that.   

That functionality _should_ be working with 9.12-MR1.   I need to understand why not.  I will investigate (start of semester here, so I'm not as responsive as I would like).

Second, the evaluation benchmarks will generate those complaints, but that's a not about DaCapo, but rather about Java versions and code versions, unfortunately, and in itself not a reason not to use evaluation.

I sincerely hope to have a new evaluation version out soon (where I've worked around some of those reflection issues, as well as adding new workloads etc etc).

--Steve

**Comment by JuliaLawall** [comment]
On Tue, 10 Aug 2021, Steve Blackburn wrote:

>
> Hi Julia,
>
> Thanks for that.
>
> That functionality should be working with 9.12-MR1. I need to understand why not. I will investigate (start of semester here, so I'm not as responsive as I would like).
>
> Second, the evaluation benchmarks will generate those complaints, but that's a not about DaCapo, but rather about Java versions and code versions, unfortunately, and in itself not a reason not to use evaluation.

OK.  Thanks for the confirmation that the new benchmarks should be
performing their correct computations despite the warning messages.

julia

**Comment by steveblackburn** [comment]
I think the root problem is that the help is less than helpful.   These options need to be used together with `-C`.

Fixed in dd3edb36831fde05d259367d4be4122f8d78425f.
